test/unit/Cover/buyCover.js

```diff
@@ -802,4 +805,464 @@ describe('buyCover', function () {
+
+    const {
+      amount,
+      productId,
+      coverAsset,
+      period,
+      expectedPremium,
+      capacity,
+      targetPriceRatio,
+      activeCover,
+      coverId,
+      segmentId,
+    } = buyCoverFixture;
+
+    // create a 2nd pool
+    await createStakingPool(
+      cover,
+      productId,
+      capacity,
+      targetPriceRatio,
+      activeCover,
+      stakingPoolManager,
+      stakingPoolManager,
+      targetPriceRatio,
+    );
+
+    // create a 3er pool
+    await createStakingPool(
+      cover,
+      productId,
+      capacity,
+      targetPriceRatio,
+      activeCover,
+      stakingPoolManager,
+      stakingPoolManager,
+      targetPriceRatio,
+    );
+
+    // create a 4th pool
+    await createStakingPool(
+      cover,
+      productId,
+      capacity,
+      targetPriceRatio,
+      activeCover,
+      stakingPoolManager,
+      stakingPoolManager,
+      targetPriceRatio,
+    );
+
+    // create a 5th pool
+    await createStakingPool(
+      cover,
+      productId,
+      capacity,
+      targetPriceRatio,
+      activeCover,
+      stakingPoolManager,
+      stakingPoolManager,
+      targetPriceRatio,
+    );
+
+    const globalRewardsRatio = 5000;
+    await cover.connect(gv1).updateUintParameters([1], [globalRewardsRatio]);
+
+    const coverAmountInAsset = amount.div(5);
+
+    const stakingPool1Before = await tokenController.stakingPoolNXMBalances(0);
+    const stakingPool2Before = await tokenController.stakingPoolNXMBalances(1);
+    const stakingPool3Before = await tokenController.stakingPoolNXMBalances(2);
+    const stakingPool4Before = await tokenController.stakingPoolNXMBalances(3);
+    const stakingPool5Before = await tokenController.stakingPoolNXMBalances(4);
+
+    await cover.connect(coverBuyer).buyCover(
+      {
+        owner: coverReceiver.address,
+        productId,
+        coverAsset,
+        amount,
+        period,
+        maxPremiumInAsset: expectedPremium,
+        paymentAsset: coverAsset,
+        payWitNXM: false,
+        commissionRatio: parseEther('0'),
+        commissionDestination: AddressZero,
+        ipfsData: '',
+      },
+      [
+        { poolId: 0, coverAmountInAsset },
+        { poolId: 1, coverAmountInAsset },
+        { poolId: 2, coverAmountInAsset },
+        { poolId: 3, coverAmountInAsset },
+        { poolId: 4, coverAmountInAsset },
+      ],
+      {
+        value: expectedPremium,
+      },
+    );
+
+    const stakingPool1After = await tokenController.stakingPoolNXMBalances(0);
+    const stakingPool2After = await tokenController.stakingPoolNXMBalances(1);
+    const stakingPool3After = await tokenController.stakingPoolNXMBalances(2);
+    const stakingPool4After = await tokenController.stakingPoolNXMBalances(3);
+    const stakingPool5After = await tokenController.stakingPoolNXMBalances(4);
+
+    // validate that rewards increased
+    expect(stakingPool1After.rewards).to.be.gt(stakingPool1Before.rewards);
```

Comment:
let's make those assertions stronger with the amount that's expected to be minted for each as in the previous test.